tmdb/show.go

```diff
@@ -568,7 +601,7 @@ func (show *Show) SetArt(item *xbmc.ListItem) {
 		}
 	}
 
-	item.Thumbnail = item.Art.Poster
+	item.Thumbnail = item.Art.Thumbnail
```

Comment:
I think there was a case that `Thumbnail` was not available for `Show` in TMDB. It is present now?

Reply:
now we use Backdrop as thumbnail.
`item.Art.Thumbnail = ImageURL(show.BackdropPath, thumbnailQuality)`

but maybe it is possible that some very new show/movie does not have backdrop, but has poster.
i can add a check for this edge case.

Reply:
I feel there is a general need in fallback art selection implemented.

Like `if we don't have art of type A, but we have art of type B, then use type B`. 
So that we won't have empty arts.

@matke-84 What do you think?

Reply:
added a fallback for thumbnail to movie and show. season/episode will use this fallback logic because of cascade calls.

Reply:
@elgatito
I think it's not necessary. This is mostly done by the skin. Today, most skinners use variables for artworks.
Also if one day we will use fanart.tv as a fallback for tmdb artworks. For example, about thumbs, which we mention here. If there is no backdrop in the tmdb database, which can happen, now the poster would be displayed with last change by @antonsoroko. But if only the tmdb backdrop remained, maybe fanart.tv would have it in its database and then it would appear. Is this good logic?

Reply:
@matke-84 I think elgatito meant this:
Should we try to use tmdb poster instead of tmdb backdrop for every Kodi art that we usually set to backdrop (fanart, landscape, thumbnail)?
And maybe vice versa: if we have only tmdb backdrop - should we set Kodi poster to tmdb backdrop? 

In my opinion - this is unnecessary logic for rare edge case and it will make our art logic too complicated, but if it not a rare case when tmdb does not have some poster/backdrop - then we can implement this logic.

Reply:
@antonsoroko 
I still think that should be handled by the skin. 

> Should we try to use tmdb poster instead of tmdb backdrop for every Kodi art that we usually set to backdrop (fanart, landscape, thumbnail)?
> And maybe vice versa: if we have only tmdb backdrop - should we set Kodi poster to tmdb backdrop?

Also, I don't think this is good. Poster and backdrop are two different forms of artwork. We are talking about a poster that is a portrait image and a backdrop that is a landscape image. Fitting a poster into a landscape container and vice versa is never good.

Reply:
@matke-84  I would mostly agree, but. There are things skin can deal with and improve. 
But my main target is to make it work properly with default skin (Estuary), so that you would see a poster, thumbnail, fanart for Movie/Episode/Show/Season. 

So if there is something we can easily set as a fallback art to make default skin show all the arts (yes, it is very limited, so should not be a problem).

Reply:
@elgatito please see commit [Use fallback art for poster/fanart/thumbnail](https://github.com/elgatito/elementum/pull/111/commits/390b78deeed94a2cab1ceff9507eafb91eed6182) 

movie/show:
poster uses poster or backdrop
backdrop uses backdrop or poster
season:
get art from show, then try to use its own art

episode:
get art from season, then set Thumbnail to Still.
If poster or backdrop is missing - set to Still.

---

i hope this covers all cases.

Reply:
@matke-84 I would happily accept all those setters if @matke-84 gives his feedback after testing 👍

Reply:
@elgatito @antonsoroko 
Of course. Only if it is not a problem for someone to compile the win64 version. I will test it when I get home.